vendor/engines/saas/app/jobs/login_private_job.rb

```diff
@@ -1,7 +1,7 @@
 class LoginPrivateJob < ActiveJob::Base
 
   def perform(params)
-    if params['emails'].respond_to?('detect')
+    if params['emails'].is_a?(Array)
       email = params['emails'].detect{|e| e['primary'] == true}
       params['user']['email'] = email['email']
```

Comment:
This will throw if `email` is an empty array, I suppose we can do something like `params['emails'].is_a?(Array) && params['emails'].size`